File: Python/Recursions/count_vowel.py
def check_vowel(char):
    if char in "aeiou":
        return 1
    return 0


def count_vowel(str, str_len):
    if str_len == 1:
        return check_vowel(str[0])
    return count_vowel(str, str_len - 1) + check_vowel(str[str_len - 1])


s = "educativeio"
print(count_vowel(s, len(s)))

File: JavaScript/Recursions/countVowel.js
const checkVowel = (n) => {
    const vowels = "aeiou";

    for(let i = 0; i < vowels.length; i++) {
        if(vowels[i] === n) {
            return 1;
        }
    }
    return 0;
}

const countVowel = (str, strLen) => {
    if(strLen === 1) {
        return checkVowel(str[0]);
    }
    return countVowel(str, strLen-1) + checkVowel(str[strLen-1]);
}

const s = "educativeio";
console.log(countVowel(s, s.length));